bundles/action/org.openhab.action.pushsafer/src/main/java/org/openhab/action/pushsafer/internal/Pushsafer.java

```diff
@@ -0,0 +1,174 @@
+
+            // Private or Alias Key
+            if (!StringUtils.isEmpty(apiKey)) {
+                addEncodedParameter(data, MESSAGE_KEY_API_KEY, apiKey);
+            } else {
+                logger.error("Private or Alias key not specified.");
+                return false;
+            }
+
+            // Message
+            if (!StringUtils.isEmpty(message)) {
+                if ((message.length() + title.length()) <= API_MAX_MESSAGE_LENGTH) {
+                    addEncodedParameter(data, MESSAGE_KEY_MESSAGE, message);
+                } else {
+
+                    logger.error("Together, the event message and title total more than {} characters.",API_MAX_MESSAGE_LENGTH);
+                    return false;
+                }
+            } else {
+                logger.error("The event message is missing.");
+                return false;
+            }
+
+            // Device
+            if (!StringUtils.isEmpty(device)) {
+                addEncodedParameter(data, MESSAGE_KEY_DEVICE, device);
+            }
+
+            // Title
+            if (!StringUtils.isEmpty(title)) {
+                addEncodedParameter(data, MESSAGE_KEY_TITLE, title);
+            }
+
+            // Icon
+            if (!StringUtils.isEmpty(icon)) {
+                addEncodedParameter(data, MESSAGE_KEY_ICON, icon);
+            }
+
+            // Vibration
+            if (!StringUtils.isEmpty(vibration)) {
+                addEncodedParameter(data, MESSAGE_KEY_VIBRATION, vibration);
+            }
+
+            // Sound
+            if (!StringUtils.isEmpty(sound)) {
+                addEncodedParameter(data, MESSAGE_KEY_SOUND, sound);
+            }
+
+            String content = data.toString();
+
+            logger.debug("Executing post with the following content: {}",content);
+            String response = HttpUtil.executeUrl("POST", API_URL, IOUtils.toInputStream(content), CONTENT_TYPE, timeout);
+            logger.debug("Raw response: {}",response);
+
+            try {
+                if (StringUtils.isEmpty(response)) {
+                    logger.error(
+                        "Received an empty response from our Pushsafer API call. This can mean either we are having trouble connecting to the Pushsafer API or the Pushsafer API is actively enforcing rate limits with a connection time-out.");
+                    return false;
+                }
+
+                if (StringUtils.containsIgnoreCase(response, API_RETURN_STATUS_SUCCESS)) {
+                    return true;
+                } else {
+
+                    logger.error("Received error message from Pushsafer: {}",response);
+                    return false;
+                }
+            } catch (Exception e) {
+
+                logger.warn("Can't parse response from Pushsafer: {}",response);
+                return false;
+            }
+        } catch (Exception e) {
+
+            logger.error("An error occurred while notifying your mobile device.");
```

Comment:
At least the message from the exception should be included in this error log output.